controllers/nutanixmachine_controller.go

```diff
@@ -906,3 +897,46 @@ func (r *NutanixMachineReconciler) isGetRemoteClientConnectionError(err error) b
 	const expectedErrString = "connect: connection refused"
 	return strings.Contains(err.Error(), expectedErrString)
 }
+
+func (r *NutanixMachineReconciler) GetSubnetAndPEUUIDs(rctx *nctx.MachineContext) (string, []string, error) {
+	if rctx == nil {
+		return "", nil, fmt.Errorf("cannot create machine config if machine context is nil")
+	}
+	log := ctrl.LoggerFrom(rctx.Context)
+	if rctx.Machine.Spec.FailureDomain == nil || *rctx.Machine.Spec.FailureDomain == "" {
+		log.V(1).Info("no failure domain found on machine. Directly searching for Prism Element cluster")
+		if rctx.NutanixMachine.Spec.Cluster.Name == nil && rctx.NutanixMachine.Spec.Cluster.UUID == nil {
+			return "", nil, fmt.Errorf("cluster name or uuid must be passed if failure domain is not configured")
+		}
+		if len(rctx.NutanixMachine.Spec.Subnets) == 0 {
+			return "", nil, fmt.Errorf("subnets must be passed if failure domain is not configured")
+		}
+		peUUID, err := GetPEUUID(rctx.Context, rctx.NutanixClient, rctx.NutanixMachine.Spec.Cluster.Name, rctx.NutanixMachine.Spec.Cluster.UUID)
+		if err != nil {
+			return "", nil, err
+		}
+		subnetUUIDs, err := GetSubnetUUIDList(rctx.Context, rctx.NutanixClient, rctx.NutanixMachine.Spec.Subnets, peUUID)
+		if err != nil {
+			return "", nil, err
+		}
+		return peUUID, subnetUUIDs, nil
+	}
+
+	log.V(1).Info("failure domain config found. Ignoring cluster config on machine object (if any)")
+
+	failureDomainName := *rctx.Machine.Spec.FailureDomain
+	failureDomain, err := GetFailureDomain(failureDomainName, rctx.NutanixCluster)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to find failure domain %s", failureDomainName)
+	}
+	cUUID, err := GetPEUUID(rctx.Context, rctx.NutanixClient, failureDomain.Cluster.Name, failureDomain.Cluster.UUID)
```

Comment:
consider renaming to `peUUID` for clarity.

Reply:
Renamed